torch_geometric/nn/models/schnet.py

```diff
@@ -224,17 +225,34 @@ def from_qm9_pretrained(root: str, dataset: Dataset, target: int):
 
         return net, (dataset[train_idx], dataset[val_idx], dataset[test_idx])
 
-    def forward(self, z, pos, batch=None):
-        """"""
+    @staticmethod
+    def _validate_fwd_args(z, pos, batch, edge_index, edge_weight):
         assert z.dim() == 1 and z.dtype == torch.long
+
+        if batch is not None:
+            assert batch.dim() == 1 and batch.dtype == torch.long
+
+        if pos is not None:
+            assert edge_index is None and edge_weight is None
+        else:
+            assert edge_index is not None and edge_weight is not None
+
+    def forward(self, z, pos=None, batch=None, edge_index=None,
+                edge_weight=None):
+        """"""
```

Comment:
Add a short doc explaining which variables can and cannot be `None` at the same time. Also add that if `edge_index` and `edge_weight` are not given it will be computed in every forward pass.